src/test/strencodings_tests.cpp

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <test/util/setup_common.h>
+#include <util/strencodings.h>
+
+#include <boost/test/unit_test.hpp>
+
+#include <array>
+#include <string>
+#include <string_view>
+
+BOOST_AUTO_TEST_SUITE(strencodings_tests)
+
+BOOST_AUTO_TEST_CASE(util_HexStr)
```

Comment:
Mind that we already have a test for HexStr in `src/test/util_tests.cpp`. Might want to remove that one, or at least move it into here.

Reply:
ah, I didn't see that! I'll move the test over that checks each character and remove the rest